modules/custom/d_product/src/Plugin/views/argument/NodeVid.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\d_product\Plugin\views\argument;
+
+
+use Drupal\node\Plugin\views\argument\Vid;
+
+/**
+ * Argument handler to accept a node revision id.
+ *
+ * @ViewsArgument("node_vid")
+ */
+class NodeVid extends Vid {
+
+  /**
+   * Override the behavior of title(). Get the title of the revision.
+   */
+  public function titleQuery() {
+    $titles = [];
+
+    $results = $this->database->query('SELECT nr.vid, nr.nid, nfr.title 
+      FROM {node_revision} nr
+      INNER JOIN {node_field_revision} nfr ON nr.vid = nfr.vid
+      WHERE nr.vid IN ( :vids[] )', array(':vids[]' => $this->value))->fetchAllAssoc('vid', \PDO::FETCH_ASSOC);
+    $nids = [];
+    foreach ($results as $result) {
+      $nids[] = $result['nid'];
+    }
+
+    $nodes = $this->nodeStorage->loadMultiple(array_unique($nids));
```

Comment:
w sumie to niepotrzebne byłoby `array_unique`, gdyby w foreach robić przypisanie `$nids[$result['nid']] = $result['nid'];` :)  

ponadto, sprawdz w dokumentacji, ale jest sposob na pobranie wynikow w postaci 
array [
 nid => title
]


bo z tego co widze tylko to potrzebujesz tu :thinking: 

to chyba powinno byc to:
`->fetchAllKeyed(1, 2);` - cyferki to indeksy pobieranych pol z listy `SELECT nr.vid, nr.nid, nfr.title `